watcher

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+trap ctrl_c SIGINT
+
+function ctrl_c {
+  make --no-print-directory clean && \
+    echo -e "\e[92mPatch cleaned\e[0m" || \
+    echo -e "\e[33mCould not clean patch\e[0m"
+  exit $?
+}
+
+function watch_inotify {
+  inotifywait --recursive \
+    --event modify,move,create,delete \
+    $DIRECTORY_TO_OBSERVE
+}
+
+function watch_fswatch {
+  fswatch --recursive --one-event \
+    --event Created --event Updated --event Removed \
+    ${DIRECTORY_TO_OBSERVE}
+}
+
+
+if command -v inotifywait > /dev/null 2>&1; then
+  wichWatcher=inotify
+  function watcher { watch_inotify; }
+elif command -v fswatch > /dev/null 2>&1; then
+  wichWatcher=fswatch
+  function watcher { watch_fswatch; }
+else
+  echo -e "\e[31m[ERROR] No watcher available\e[0m"
+  echo "Neither 'inotify' nor 'fswatch' are available."
+  echo "You can follow 'CONTRIBUTING.md' guidelines to install one of them,"
+  echo ""
+  echo -e "\e[33malternative:\e[0m"
+  echo "  run 'make set-override' first,"
+  echo "  then 'make patch' every time you change a file in 'srcd' dir"
+  echo "  and 'make clean' once you finish developing"
+  echo ""
+  exit 1
+fi
+
+make --no-print-directory apply-patch
```

Comment:
you listen for `delete` event but doesn't delete anything

Reply:
good catch...
I don't want to process the event to remove files on the other side...
Do you think that would it be safe calling `git clean -fd $(SUPERSET_DIR)` while the developer is working there?
(I'm not used to this `git clean`, so I'm not sure if some changes in `$(SUPERSET_DIR)` could be lost because of the `git clean`)

If it is not, I'd just exclude the delete event, and document:
>_Elements deleted in `srcd` won't be deleted from `sourced`; you can stop the watcher and call it again instead._

Reply:
I don't feel comfortable if something can remove my work without my explicit action.
So I don't think `git clean` is a good solution here.

Keeping stuff can cause SUPER WEIRD behavior. For example, a user does `import from 'path/*'` somewhere.

Can we just call `rm` if we received the event?

Reply:
Well.... `import *`... _sigh_ 

I should dig more on how the `delete` event is logged (and also the `move` event, if it's different...) considering the different platforms...

What if it could be done in a separate PR, and documented in the meantime?

Reply:
issue opened to be done separately https://github.com/src-d/sourced-ui/issues/225

Reply:
> Can we just call `rm` if we received the event?

This shouldn't be hard, and it seems sensible and the most straightforward thing to do IMHO, isn't it?

Reply:
Each watcher returns the detected changes in a different way, so it should be parsed the output in both cases.
Since it is not critical, and it's documented, I think we can merge and improve later,

Reply:
Okay! Thanks for the explanation!